launch/pipeline/service.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+from typing import Any, Callable, Dict, List, Optional
+
+from launch.pipeline.deploy import Deployment
+from launch.pipeline.runtime import Runtime
+
+
+class ServiceDescription:
+    """
+    Base ServiceDescription.
+    """
+
+    def call(self, *args, **kwargs):
+        raise NotImplementedError()
+
+
+class SingleServiceDescription(ServiceDescription):
+    """
+    The description of a service.
+    """
+
+    def __init__(
+        self,
+        service: Callable,
+        runtime: Runtime,
+        deploy: Deployment,
+        init_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__()
+        self.service = service
+        self.runtime = runtime
+        self.deploy = deploy
+        self.init_kwargs = init_kwargs
+
+        self._callable_service: Optional[Callable] = None
+
+    def _init_callable_service(self):
+        if self._callable_service:
+            return
+        if inspect.isclass(self.service):
+            kwargs = self.init_kwargs or {}
+            self._callable_service = self.service(**kwargs)
+        else:
+            assert (
+                not self.init_kwargs
+            ), "`init_kwargs` is given, but the service is not a class"
+            self._callable_service = self.service
+
+    def call(self, *args, **kwargs):
+        self._init_callable_service()
+        return self._callable_service(*args, **kwargs)
+
+
+class SeqPipelineServiceDescription(ServiceDescription):
```

Comment:
I'd just call this `SequentialPipelineDescription`:
1. I'd prefer to write things out instead of using abbreviations if we can help it
2. I think "pipeline" already implies a pipeline of services (though I could see arguments for calling this thing either a "PipelineService" or a "ServicePipeline")